PublishParasoftResults/CoverageReportService.ts

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright 2024 Parasoft Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as fs from 'fs';
+import * as sax from 'sax';
+
+type CoberturaReport = {
+    lineRate: number;
+    linesCovered: number;
+    linesValid: number;
+    version: string;
+    packages: Map<string, CoberturaPackage>;
+}
+
+type CoberturaPackage = {
+    name: string;
+    lineRate: number;
+    classes: Map<string, CoberturaClass>;
+}
+
+type CoberturaClass = {
+    fileName: string;
+    name: string;
+    lineRate: number;
+    classId: string;
+    lines: CoberturaLine[];
+}
+
+type CoberturaLine = {
+    lineNumber: number;
+    lineHash: string;
+    hits: number;
+}
+
+export class CoverageReportService {
+    private readonly destFolderPath: string;
+
+    constructor(defaultWorkingDirectory: string) {
+        this.destFolderPath = defaultWorkingDirectory;
```

Comment:
The value of defaultWorkingDirectory can be found by tl.getVariable('System.DefaultWorkingDirectory'), so it is not necessary to pass the value through constructor parameter.